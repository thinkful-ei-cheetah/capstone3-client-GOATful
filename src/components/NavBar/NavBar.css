header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    box-shadow: 0 4px 2px -2px rgba(210, 210, 210, .6);
    background: white;
    width: 100%;
    height: 60px;
    top: 0;
    z-index: 10;
    position: relative;
}
.app-title{
  margin: auto 0;
}

.root-link {
  display: flex;
}

.app-title a{
  text-decoration: none;
  color: var(--primary);
}
.app-title a:hover{
  color: var(--Green);
  text-decoration: underline;
}
nav {
    /* position: fixed; */
    right: 0;
    font-size: 22px;
    top: 18px;
}
nav a{
    margin-right: 25px;
    text-decoration: none;
    /* color: var(--primary); */
    font-weight: 800;
}
nav a:hover{
  color: var(--Green);
  text-decoration: underline;
}
.logo {
  width: 40px;
  height: 40px;
  margin: auto 5px;
  margin-top: 5px;
  font-weight: bolder;
}
.root-link:hover {
  fill: rgb(196, 48, 43);
  transition: .5s;

}
