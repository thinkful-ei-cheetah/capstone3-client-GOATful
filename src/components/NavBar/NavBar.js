import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import UserContext from '../../contexts/UserContext'
import Logo from '../Logo/Logo'
import './NavBar.css'

class NavBar extends Component {
  static contextType = UserContext

  handleLogoutClick = () => {
    this.context.processLogout()
  }

  renderLogoutLink() {
    return (
      <div>
        <span>
          {this.context.user.name}
        </span>
        <nav>
          <Link
            onClick={this.handleLogoutClick}
            to='/'>
            Logout
          </Link>
        </nav>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <nav>
        <Link to='/login'>Login</Link>
        {' '}
      </nav>
    )
  }

  render() {
    return (
      <header>
        <div className="title-container">
            <Link className="root-link" to='/'>
              <Logo className="logo" />
              <h2 className='app-title'>Laconic</h2>
            </Link>
        </div>
     
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </header>
    );
  }
}

export default NavBar
