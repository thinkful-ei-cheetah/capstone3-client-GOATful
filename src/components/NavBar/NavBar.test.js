import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import NavBar from './NavBar';
import { BrowserRouter } from 'react-router-dom'

describe('Navbar Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
          < NavBar />
        </BrowserRouter>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><NavBar/></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})