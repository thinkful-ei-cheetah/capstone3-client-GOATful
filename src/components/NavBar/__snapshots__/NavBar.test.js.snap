// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders without crashing 1`] = `
<header>
  <div
    className="title-container"
  >
    <Link
      className="root-link"
      to="/"
    >
      <Logo
        className="logo"
      />
      <h2
        className="app-title"
      >
        Laconic
      </h2>
    </Link>
  </div>
  <nav>
    <Link
      to="/login"
    >
      Login
    </Link>
     
  </nav>
</header>
`;
